### DISCUSSION TITLE
EnumMembers are not bounded [13745]

### DISCUSSION THREAD

**Discussion by ctefer** [eProsima/IDL-Parser#93] [Ideas]
Topic Types generated from IDL that contain enums are not bounded. The instantiation of these types only calls the annotation constructor of the member class and there's no chance to set the type code. I'm not sure if this is done on purpose due to serialization or if there's no way to bound enums. 

Would it be possible to set each EnumMember typecode to the parent EnumTypeCode? I believe this just results in the default bounding being true.

Is there an issue with serializing the enums that prevents this bounding? I wouldn't think so as C++ doesn't really care about enum types if they are implemented as a standard integer (which I believe is being done when the pubsubtypes are generated).

**Comment by JLBuenoLopez** [comment]
Hi @ctefer 

Sorry for the late reply. [XTypes v1.3 specification](https://www.omg.org/spec/DDS-XTypes/1.3/About-DDS-XTypes) establishes in clause **7.3.1.2.1.5 Enumerated Literal Values** the following:

> Prior to this specification, it was impossible to indicate that objects of enumerated types could be stored using an integer size other than 32 bits. This specification uses the @bit_bound annotation defined in Sub Clause 8.3.4.1 of [IDL] for this purpose.

> It is important to note that the value member of the annotation may take any value from 1 to 32, inclusive, when this annotation is applied to an enumerated type.

So the enumeration size can be actually be configured. However I am afraid that this annotation is not yet supported for enumerations. Thus, I am labeling this ticket as an enhancement and moving it to the Ideas discussion forum where enhancements and feature requests are discussed.
